### PR TITLE
CNV 2.1 upgrade + install docs

### PR DESCRIPTION
**Pull request by ousleyp**
Note: The upgrade docs depend on/overlap with the installation guide updates for 2.1. For now, I'm bundling all the updates in this one PR, but if that turns out to be counterproductive, I'll split them up.

Relevant Jiras: CNV-2278, CNV-2473, CNV-2581

**Preview builds**

Upgrade: http://file.bos.redhat.com/pousley/100319/cnv-2.1-upgrade/cnv/cnv_install/upgrading-container-native-virtualization.html

Install: http://file.bos.redhat.com/pousley/100319/cnv-2.1-upgrade/cnv/cnv_install/installing-container-native-virtualization.html

Uninstall: http://file.bos.redhat.com/pousley/100319/cnv-2.1-upgrade/cnv/cnv_install/uninstalling-container-native-virtualization.html

PTAL, @openshift/team-documentation - thanks!

Please cherrypick to enterprise-4.2 only.

### CODE REVIEW

**Comment by ousleyp** [comment]
Attn @nellyc @rthallisey - I wrote notes in the relevant Jiras but this is ready for review. I am concerned that I might not have fully understood the google docs that were provided, so please let me know if anything is inaccurate. Thanks!

File: cnv/cnv_install/upgrading-container-native-virtualization.adoc
Comment (stu-gott) [inline cnv/cnv_install/upgrading-container-native-virtualization.adoc]: I'm uncertain of the phrasing here. How about "Automatic updates are enabled during installation"? or "can be" if it's optional.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (stu-gott) [inline modules/cnv-about-upgrading-cnv.adoc]: you must restart or live migrate the virtual machine.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (stu-gott) [inline modules/cnv-about-upgrading-cnv.adoc]: Suggest: DataVolumes and their associated PersistentVolumeClaims

**Review by stu-gott — COMMENTED** [review]
Nice job!

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: Good catch, thanks! Fixed.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: Sounds good to me. I made the change. Thanks!

File: cnv/cnv_install/upgrading-container-native-virtualization.adoc
Comment (ousleyp) [reply cnv/cnv_install/upgrading-container-native-virtualization.adoc]: It does sound strange, but a lot of our content includes "You <verb> <the thing>" phrasing. We try to avoid the passive voice, and "can" is sometimes confusing/unclear.

I am discussing this via email with Steve G and Ryan, because I think we don't *want* it to be optional, but it currently is optional in the UI. I'll leave this comment open and update the phrasing when we come to a conclusion. :)

(Thanks for the review, @stu-gott!)

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: I would drop "in the web console" and replace to:  Navigate to the *Operators → OperatorHub* page. or In the left menu navigate to the ...

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: s/navigate to /log in

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: why a capital case for CNVProductName here? It's not a beginning of the sentence. For example, a paragraph above, before "Prerequisites" has it with low case + Operator.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: This option is not always available or supported maybe? Please double check with @rthallisey maybe.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: @rthallisey now there is only "alpha" channel. Will it be changed to "2.1"?

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: I would also mention how it works with "Manual" option.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (slintes) [inline modules/cnv-about-upgrading-cnv.adoc]: Just curious, do people know what the `virt-launcher` pod is (the pod which actually runs the VMI)?
But I also have no better idea to describe this :)

File: modules/cnv-preparing-to-install.adoc
Comment (slintes) [inline modules/cnv-preparing-to-install.adoc]: IMHO this page is obsolete. The creation of the namespace / project is also mentioned on the `Subscribing to the KubeVirt HyperConverged Cluster Operator catalog` page. The prerequisites and procedure from this page can go there as well. WDYT?

**Review by alexxa — CHANGES_REQUESTED** [review]

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Ah, good catch! I think I misinterpreted this guideline: https://github.com/ousleyp/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization

I missed the "defer to the product's capitalization style" part (not to mention that I wasn't consistent about this). I will change it back to lower case.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: I agree it's a good idea to add a little more context here, so I'll try to include your brief description. Thanks for catching that!

File: modules/cnv-preparing-to-install.adoc
Comment (ousleyp) [reply modules/cnv-preparing-to-install.adoc]: It is just a single command at this point, but I'm not sure if we should be switching between the terminal and the UI in the same procedure. I can't have two procedures in one module (both of these files that you referred to are modules, and they are arranged in an assembly that encompasses the entire user story). Thus, I think I might keep this in, at least until it becomes possible to create "openshift-*" namespaces from the UI. I'll consider it further, though. Thanks!

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Good catch, that was leftover from before the first step existed.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Good call, @alexxa 
@rthallisey do you think it makes sense to change this to just say:
"This option is not supported for use with container-native virtualization." ? I don't think the "This option is not always available" part is very helpful anyway.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: My impression is that "Manual" is not actually supported. I'm waiting on @rthallisey 's response to an email from Steve G. It sounded like the way the options are displayed in the UI might be changed so that automatic is the default. I'll rephrase this when I hear more.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: @slintes Do you think this is accurate?
"Each virtual machine has a `virt-launcher` Pod that runs the virtual machine
instance. The `virt-launcher` Pod contains an instance of `libvirt`, which is
used to manage the virtual machine process."

Comment (slintes) [reply modules/cnv-about-upgrading-cnv.adoc]: I'd prefer `The virt-launcher Pod *runs* an instance...`, otherwise :+1:

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (slintes) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Are you sure? To my best knowledge in Openshift we talk about projects. Technically they are the same though. I would align this with the rest of the Openshift docs

Comment (alexxa) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: yes, there is a [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1732932)

Comment (slintes) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: oh, interesting, thx for clarification :+1:

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (rthallisey) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: I think we want to limit the deploy to `openshift-cnv` since it's how QE tested and some of CNV's operator are singletons.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (rthallisey) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: We don't need to mention Manual, but rather something like "use the default option". 

`Use the default upgrade policy, *Automatic*`

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: Sounds good, I'll make that change now.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: @rthallisey Okay, so then the way it is currently phrased seems correct. The step instructs the user to "Select *A specific namespace on the cluster* from the *Installation Mode*
list and choose the `openshift-cnv` namespace." It describes the two options but says that the "all namespaces" option is not supported for use with CNV. 

Or did you mean that the other installation mode option is going to be removed from the UI?

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [inline modules/cnv-subscribing-to-hco-catalog.adoc]: s/{CNVProductName}/KubeVirt HyperConverged Cluster
why it was replaces  wise verse at first place?
image

**Review by alexxa — CHANGES_REQUESTED** [review]
@ousleyp please see a few more suggestions. Then it should be good to merge.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (slintes) [reply modules/cnv-about-upgrading-cnv.adoc]: Not sure if I understand...?
IMHO it's clear that "uninterrupted" is in the context of upgrades only.
But removing that part also works.
I don't have a preference.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (rthallisey) [inline modules/cnv-monitoring-upgrade-status.adoc]: `Conditions` represent the system's best guess based on available information.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (rthallisey) [inline modules/cnv-monitoring-upgrade-status.adoc]: Monitor the aggregated status of all {CNVProductName} component `Conditions` by running

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Good question. I could have sworn that it was the Container-native Virtualization Operator in the material @rthallisey provided, but looking at it now I just see Kubevirt HyperConverged Cluster Operator. Will fix.

File: modules/cnv-about-upgrading-cnv.adoc
Comment (ousleyp) [reply modules/cnv-about-upgrading-cnv.adoc]: @alexxa I like your suggestion to remove "Because workloads are uninterrupted." I think the phrase is somewhat confusing in the context of that sentence, and the intended meaning is already conveyed through the bullet point levels.

**Comment by ousleyp** [comment]
@alexxa  I made the changes, but I'm having some issues with getting the preview build to work (the upgrade changes aren't appearing for some reason). I'll let you know when it is functional, but in the meantime, feel free to look at the latest commit. Thanks!

**Review by alexxa — APPROVED** [review]

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (rthallisey) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: I would say to only document the specific namespace method, so it's clear that's the path we want to steer the user.

**Review by kalexand-rh — COMMENTED** [review]
I have a few picks and suggestions, but it's looking good.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (alexxa) [reply modules/cnv-monitoring-upgrade-status.adoc]: may it stay as it's now. We give that command to monitor the status of the upgrade. The given output will be displayed only when the upgrade is finished, as the statement above says.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (alexxa) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: There are two of them.

Comment (kalexand-rh) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: Thanks! Then you need to present the two options in an unordered list and make it clear that you do one of them.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (kalexand-rh) [reply modules/cnv-monitoring-upgrade-status.adoc]: As long as the note's nested the right way, yes.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (ousleyp) [reply modules/cnv-monitoring-upgrade-status.adoc]: @kalexand-rh I'm not quite sure what you mean -- should I use the longer construction here? The PR comments make it seem like that is still up in the air.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (ousleyp) [reply modules/cnv-monitoring-upgrade-status.adoc]: I actually intended for it to *not* be connected to the command, because `PHASE` refers to the first part of the procedure, and `Conditions` refers to the second, optional part. Do you think that makes sense?

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (ousleyp) [reply modules/cnv-monitoring-upgrade-status.adoc]: `PHASE` is called out in the first part of the procedure. I think that `Conditions` needs to be added to the sample output for the last command.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: It seems right to me. The note is aligned with step 5a.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: This is actually supposed to be more like "Select *2.1* from the list of available *Update Channel* options."

Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: I think there's only one option, which is why I originally went with that (admittedly awkward) construction.

File: modules/cnv-subscribing-to-hco-catalog.adoc
Comment (ousleyp) [reply modules/cnv-subscribing-to-hco-catalog.adoc]: I'm a bit confused. My understanding is that we don't want the customer to think that they could choose either option -- they need to pick "A specific namespace on the cluster." I presented both options in the note in order to explain them/give context, because both options do appear in the UI. Is that sufficient or should I rewrite the step?

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (kalexand-rh) [reply modules/cnv-monitoring-upgrade-status.adoc]: It's used in 21 places in the repo now. I'd use it, and if we pick different language, it'll get updated with the others.

File: modules/cnv-monitoring-upgrade-status.adoc
Comment (kalexand-rh) [reply modules/cnv-monitoring-upgrade-status.adoc]: After getting a little more information, I think this note needs to be removed.

Comment (alexxa) [reply modules/cnv-monitoring-upgrade-status.adoc]: >> I don't see either PHASE or Conditions called out.
Phase and Conditions are not things to be called out, it's like a variable displaying the status of the things. Phase title is displayed by `oc get csv` command (value is Installing/Replacing/ Succeeded, ..), and conditions by that complicated yaml parsing command. 
>> After getting a little more information, I think this note needs to be removed.
Bring it back please. Currently, we need to stress, that e.g. even if the Phase reports "Succeeded", it may happen that not all things went/work well, therefore it's "best guess based on... "

Comment (ousleyp) [reply modules/cnv-monitoring-upgrade-status.adoc]: Opened PR #17095 to address this.

**Pull request merged**
